services/121-service/src/payments/fsp-integration/cooperative-bank-of-oromia/services/cooperative-bank-of-oromia-api.service.ts

```diff
@@ -0,0 +1,205 @@
+import { Injectable } from '@nestjs/common';
+import { AxiosResponse } from '@nestjs/terminus/dist/health-indicator/http/axios.interfaces';
+import { TokenSet } from 'openid-client';
+
+import { env } from '@121-service/src/env';
+import { CooperativeBankOfOromiaApiAuthenticationRequestBodyDto } from '@121-service/src/payments/fsp-integration/cooperative-bank-of-oromia/dtos/cooperative-bank-of-oromia-api-authentication-request-body.dto';
+import { CooperativeBankOfOromiaApiAuthenticationResponseBodyDto } from '@121-service/src/payments/fsp-integration/cooperative-bank-of-oromia/dtos/cooperative-bank-of-oromia-api-authentication-response-body.dto';
+import { CooperativeBankOfOromiaApiPaymentResponseBodyDto } from '@121-service/src/payments/fsp-integration/cooperative-bank-of-oromia/dtos/cooperative-bank-of-oromia-api-payment-response-body.dto';
+import { CooperativeBankOfOromiaApiTransferRequestBodyDto } from '@121-service/src/payments/fsp-integration/cooperative-bank-of-oromia/dtos/cooperative-bank-of-oromia-api-transfer-request-body.dto';
+import { CooperativeBankOfOromiaTransferResultEnum } from '@121-service/src/payments/fsp-integration/cooperative-bank-of-oromia/enums/cooperative-bank-of-oromia-disbursement-result.enum';
+import { CooperativeBankOfOromiaApiError } from '@121-service/src/payments/fsp-integration/cooperative-bank-of-oromia/errors/cooperative-bank-of-oromia-api.error';
+import { CooperativeBankOfOromiaApiHelperService } from '@121-service/src/payments/fsp-integration/cooperative-bank-of-oromia/services/cooperative-bank-of-oromia-api-helper.service';
+import { CustomHttpService } from '@121-service/src/shared/services/custom-http.service';
+
+@Injectable()
+export class CooperativeBankOfOromiaApiService {
+  private tokenSet: TokenSet;
+  private readonly cooperativeBankOfOromiaTransferURL: URL;
+  private readonly cooperativeBankOfOromiaAuthenticateURL: URL;
+
+  // client id and secret are used for authentication
+
+  public constructor(
+    private readonly httpService: CustomHttpService,
+    private readonly cooperativeBankOfOromiaApiHelperService: CooperativeBankOfOromiaApiHelperService,
+  ) {
+    const transferUrlPart = 'nrc/1.0.0/transfer'; // This path is the same on UAT and Production
+    if (
+      env.MOCK_COOPERATIVE_BANK_OF_OROMIA ||
+      !env.COOPERATIVE_BANK_OF_OROMIA_API_URL
```

Comment:
why this pattern? If the api_url is not specified we automatically switch to mock?